### DISCUSSION TITLE
list(String url, depth = -1) doesn't do infinite depth

### DISCUSSION THREAD

**Discussion by MeurglysIII** [lookfirst/sardine#446] [Q&A] [answered]
`List<DavResource> list(String url, int depth)` is documented to look at infinite recursion if `depth = -1`, but for me it works the same as `depth = 1`. 
Is there any way to get all files in `List<DavResource>` without writing recursion myself (it works slow)

Thank you!

**Comment by lookfirst** [comment]
If you think of Sardine as a sort of HTTP client wrapper that knows about webdav, then why don't you act like a browser and put all those requests into multiple threads and execute them concurrently?

**Reply by MeurglysIII** [reply]
Does that mean several concurrent connections to the server? How do I know how many concurrent connections are allowed and what happens when I exceed the limit? Besides the server policies might change with time. The idea is interesting none the less.
What I would like to know if it is possible to get the files list with recursion with just one quick command. Is this information even stored on the server in a ready form?

**Reply by lookfirst** [reply]
If you exceed the limit, I'd presume the server would respond with `429 Too Many Requests`.

> What I would like to know if it is possible to get the files list with recursion with just one quick command.

Through webdav? Probably not. If that was enabled, you could easily DDoS a server with a single request.

If you want that information, you could probably expose another API on your server that gets you what you want. I'd presume that this information would be cached on the server to prevent DDoS. That caching layer probably falls outside of the scope of webdav itself though since it could be implemented in a number of ways.

**Reply by MeurglysIII** [reply, marked as answer]
That makes sense, thank you. I also read that `infinite` depth might not be supported on some servers. I consider this question closed.
